### ISSUE TITLE
Undefined variable in nested lambdas

### ISSUE THREAD

**Issue by dmzkrsk** [PyCQA/pylint#1395]
### Steps to reproduce

A simplified fragment from my project

```python
def chain(x, f):
    return [x, f(x)]

def make(ms):
    return lambda x: chain(ms, lambda xs: reversed([x] + xs))
```

### Current behavior
`E0602:Undefined variable 'x'`

### Expected behavior
No error

### pylint --version output
```
pylint 1.7.0,
astroid 1.5.0
Python 3.6.0 (default, Dec 24 2016, 08:01:42)
[GCC 4.2.1 Compatible Apple LLVM 8.0.0 (clang-800.0.42.1)]
```

```
-e git+https://github.com/PyCQA/astroid@31f21cd2#egg=astroid
-e git+https://github.com/PyCQA/pylint@da2e9724#egg=pylint
```

**Comment by rogalski** [comment]
Looks like a dupe of #760.

**Comment by PCManticore** [comment]
Closed as duplicate of #760
